src/Simulation/Core/IQuantumMachineOutput.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Quantum.Runtime
+{
+    /// <summary>
+    /// Interface to access the results of a program executed in a quantum machine.
+    /// </summary>
+    public interface IQuantumMachineOutput
+    {
+
+        /// <summary>
+        /// Gets an output probabilistically.
+        /// Calls to this method are not deterministic, each call can return a different output.
+        /// </summary>
+        object GetOutput();
+
+        /// <summary>
+        /// Gets the most frequent output.
+        /// If multiple outputs have the same frequency, it gets the first one in ascending order.
+        /// </summary>
+        object GetMostFrequentOutput();
+
+        /// <summary>
+        /// Gets a histogram of outputs.
+        /// The key is the output and the value is its frequency.
+        /// </summary>
+        IDictionary<object, double> Histogram { get; }
```

Comment:
To be more precise and avoid floating-point precision issues, could `double` be an `int` that represents the exact number of occurrences of each output? If needed, the frequency could still be calculated by dividing that by the number of shots.

Reply:
After chatting with @cesarzc, it seems like the Azure Quantum service itself is returning a histogram with doubles, so this would need to be a bigger change than just updating the interface.